src/xdocs/config_regexp.xml

```diff
@@ -554,6 +554,134 @@
       </subsection>
     </section>
 
+    <section name="RegexpOnFilename">
+      <subsection name="Description">
+        <p>
+          A check that makes sure that files that match specific patterns are located in their proper folders and don't show up in the wrong ones.
+          It can also be used to verify files match specific naming patterns not covered by other checks (Ex: properties, xml, etc.).
+        </p>
+      </subsection>
+
+      <subsection name="Properties">
+        <table>
+          <tr>
+            <th>name</th>
+            <th>description</th>
+            <th>type</th>
+            <th>default value</th>
+          </tr>
+          <tr>
+            <td>folderPattern</td>
+            <td>Regular expression to match the folder path against.</td>
+            <td><a href="property_types.html#regexp">regular expression</a></td>
+            <td><code>null</code></td>
+          </tr>
+          <tr>
+            <td>fileNamePattern</td>
+            <td>Regular expression to match the file path against.</td>
+            <td><a href="property_types.html#regexp">regular expression</a></td>
+            <td><code>null</code></td>
```

Comment:
@mkordas 
We don't set the pattern fields when the class is created. They are only filled in once the `init` method is called if the fields are still null, meaning the user didn't specify any properties. This is by design so when the users specify atleast one property, all the matching are under their control for full customization and not defaulting to the 'no-space' version.